kwave/options/simulation_execution_options.py

```diff
@@ -1,81 +1,143 @@
-from dataclasses import dataclass
-from typing import Optional, Union
-
-from kwave import PLATFORM, BINARY_PATH
 from kwave.ksensor import kSensor
 from kwave.utils.checks import is_unix
+from kwave import PLATFORM, BINARY_PATH
+import logging
+from dataclasses import dataclass, field
+from typing import Optional, Union
+from pathlib import Path
+import warnings
+
+warnings.filterwarnings("always")
+warnings.simplefilter("always", DeprecationWarning)
+
+
+logger = logging.getLogger(__name__)
+
+DEPRECATION_VERSION = "0.3.7"
 
 
 @dataclass
 class SimulationExecutionOptions:
-    # Are we going to run the simulation on the GPU?
-    # Affects binary name and the way the simulation is run
-    is_gpu_simulation: bool = False
+    _is_gpu_simulation: bool = field(default=False, init=False)
+    _binary_name: Optional[str] = field(default=None, init=False)
+    _binary_path: Optional[Path] = field(default=BINARY_PATH, init=False)
 
-    # user defined location for the binary
-    binary_path: Optional[str] = BINARY_PATH
-    # user defined location for the binary
-    binary_name: Optional[str] = None
-    # user defined number of threads
+    is_gpu_simulation: bool = False
```

Comment:
Feels unconventional to have a parameter and a property-method with the same name.